### PR TITLE
toml to json now returns error reasons to byond + rust-analyzer.cargo.target gnu > msvc

### LINKED ISSUE #86: Why is the Rust Analyzer target set to i686-pc-windows gnu?
https://github.com/tgstation/rust-g/blob/274fb1836d83836d8b613cf300435d75870d7a1b/.vscode/settings.json#L2

Title, this has bothered contributors besides myself, who have to tell git to ignore our changes to make it the msvc target instead.

### PR DESCRIPTION
**Pull request by tralezab**
toml to json: One half of better error reporting for things like banned word filters in game, the other half has to be displaying the error on DM side.

rust-analyzer.cargo.target: fixes #86 and let me say holy shit that was one hellish part of getting the analyzer working

BLAZING
LY FAST

### CODE REVIEW

**Comment by Mothblocks** [comment]
Right now this is going to cause a backwards incompatibility issue where `rustg_read_toml_file` does something different now

I think you should change dmsrc to be like this:

```dm
#define rustg_read_toml_file_result(path) json_decode(call(RUST_G, "toml_file_to_json")(path) || "null")

/proc/rustg_read_toml_file(path)
    var/output = rustg_read_toml_file_result(path)
    if (output["success"])
        return output["content"]
    else
        CRASH(output["content"])
```

That way existing consumers and consumers that just want to runtime instead of need to manage it can use the existing thing, and anything that wants to especially handle the error case can use `rustg_read_toml_file_result`

**Comment by tralezab** [comment]
Aye aye, cap'n!

**Comment by Mothblocks** [comment]
@tralezab Needs to be in dmsrc

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-    "rust-analyzer.cargo.target": "i686-pc-windows-gnu"
+    "rust-analyzer.cargo.target": "i686-pc-windows-msvc"
```
Comment (ZeWaka) [inline .vscode/settings.json:2]: unlikely spacemaniac will accept this, per the issue

Comment (Mothblocks) [reply .vscode/settings.json:2]: he didnt comment on the issue though

Comment (AffectedArc07) [reply .vscode/settings.json:2]: This applies to literally everyone else though, and we shouldn't impact the majority for the sake of one person.

**Comment by AffectedArc07** [comment]
> Right now this is going to cause a backwards incompatibility issue where `rustg_read_toml_file` does something different now
> 
> I think you should change dmsrc to be like this:
> 
> ```dm
> #define rustg_read_toml_file_result(path) json_decode(call(RUST_G, "toml_file_to_json")(path) || "null")
> 
> /proc/rustg_read_toml_file(path)
>     var/output = rustg_read_toml_file_result(path)
>     if (output["success"])
>         return output["content"]
>     else
>         CRASH(output["content"])
> ```
> 
> That way existing consumers and consumers that just want to runtime instead of need to manage it can use the existing thing, and anything that wants to especially handle the error case can use `rustg_read_toml_file_result`

Make it `rustg_raw_read_toml_file()`, `result` is questionable.

File: .vscode/settings.json (hunk 1 above)
Comment (tralezab) [reply .vscode/settings.json:2]: this made it so much more painful to start guys

**Comment by tralezab** [comment]
👍

**Comment by tralezab** [comment]
I don't know what's up with the linux build...

**Comment by willox** [comment]
Looks like it is encoding the `enabled` field as `1` instead of `true`.

**Comment by optimumtact** [comment]
you need to fix the unit tests to expect the backtrace

/test/proc/check_toml_file2json ------------------------------------------

**Comment by tralezab** [comment]
Now I understand what's up I got it

**Pull request merged**